app/src/ui/add-repository/add-existing-repository.tsx

```diff
@@ -81,18 +103,39 @@ export class AddExistingRepository extends React.Component<IAddExistingRepositor
     this.checkIfPathIsRepository(path)
   }
 
+  private pathExists(directory: string): Promise<boolean> {
+    return new Promise<boolean>((resolve, reject) => {
+      try {
```

Comment:
Since `FS.Stat` is an asynchronous method we don't have to be afraid of it throwing